src/lib/logical_query_plan/union_node.cpp

```diff
@@ -45,6 +45,26 @@ bool UnionNode::is_column_nullable(const ColumnID column_id) const {
   return left_input()->is_column_nullable(column_id) || right_input()->is_column_nullable(column_id);
 }
 
+std::shared_ptr<LQPUniqueConstraints> UnionNode::unique_constraints() const {
+  switch (set_operation_mode) {
+    case SetOperationMode::Unique:
+    case SetOperationMode::Positions:
+      // UnionPositions merges two reference tables with the same original table(s). Any duplicate RowIDs are
+      // filtered out. As a consequence, existing unique constraints from input tables can be forwarded.
```

Comment:
This comment does not apply to Unique

Reply:
I remember the discussion we had. How about?
```
 // UnionPositions merges two reference tables with the same original table(s), computing a Set Union of their 
 // pos lists. Therefore, unique constraints remain valid.
```

Reply:
I was just hinting that because of 
```
    case SetOperationMode::Unique:
    case SetOperationMode::Positions: 
```
it somewhat looks as if it applies to both.

Reply:
`SetOperationMode::Unique` is not implemented for UniqueNode yet. But if it was, it should offer even stronger guarantees than `SetOperationMode::Positions`.

From my understanding it works as follows:

| idx | left_table | right_table | SetOperation::Positions | SetOperationMode::Unique |
| --- | ---------- | ----------- | ----------------------- | ------------------------ |
| 0   | RowID{0}   | RowID{0}    | RowID{0}                | RowID{0}                 |
| 1   | RowID{0}   | RowID{0}    | RowID{0}                | RowID{1}                 |
| 2   | RowID{1}   | RowID{1}    | RowID{1}                | RowID{2}                 |
| 3   | RowID{2}   | RowID{3}    | RowID{2}                | RowID{3}                 |
|     |            |             | RowID{3}                |                          |

In terms of unique constraints, both `SetOperation::Positions` and `SetOperationMode::Unique` should behave the same. Or am I missing something?

Reply:
This one is actually more complicated than I first thought. The "Input tables should have the same constraints." assertion only holds true for UnionPositions. For the Unique mode, which is not fully supported yet, we MIGHT be able to forward constraints if they are given on both sides. We can certainly add a constraint that goes over all columns. Honestly, as actual unions are not supported, I'd simply add a Fail() explaining that this is TODO.

Reply:
Makes sense. I add a Fail with a reference to this discussion.